pkg/export/setup/setup.go

```diff
@@ -90,130 +90,134 @@ func Global() *export.Exporter {
 	return globalExporter
 }
 
-// FromFlags returns a constructor for a new exporter that is configured through flags that are
-// registered with the given application. The constructor must be called after the flags
-// have been parsed.
-//
-// NOTE(bwplotka): This method should only setup flags, no extra logic should be done here
-// as we don't have a logger ready, and nothing was logged for the binary yet.
-// Potential risky logic can be moved to the returned function we return here.
-// See b/344740239 on how hard is to debug regressions here.
-func FromFlags(a *kingpin.Application, userAgentProduct string) func(context.Context, log.Logger, prometheus.Registerer) (*export.Exporter, error) {
-	var (
-		metadataFetchTimeout time.Duration
-		opts                 export.ExporterOpts
-	)
-	opts.UserAgentProduct = userAgentProduct
-
+func ExporterOptsFlags(a *kingpin.Application, opts *export.ExporterOpts) {
```

Comment:
> @pintohutch [`7fe3d0b` (#974)](https://github.com/GoogleCloudPlatform/prometheus-engine/pull/974/commits/7fe3d0b10a0a0258d55440ef27bc473102e02fe1#r1661194668)
> suggestion: It feels brittle to rely on opts being .Default()ed before this is called every time.
>
> What if we did the defaulting of opts in this routine, like before? Tying the flag/config definitions and default behavior centrally makes a lot of sense to me.
> 
> I think this also comes back to my previous comment of around how we envision clients handling the ExporterOpts struct.

That makes a lot of sense, but I'd still like the `Default` method so that unit tests can still use defaults without setting up flags or hardcoding them. To my shock, we were doing the former so this change prevents this. :)